File: selections-loader/backend/MarketRadio.SelectionsLoader/Domain/Task.cs
using System;
using MarketRadio.SelectionsLoader.Domain.Abstractions;

namespace MarketRadio.SelectionsLoader.Domain
{
    public class Task : Entity
    {
        public string Name { get; set; }
        public bool IsFinished { get; set; }
        public int Priority { get; set; }
        public DateTime CreateDate { get; set; }
        public DateTime? FinishDate { get; set; }
        public Guid TaskObjectId { get; set; }
        public TaskType TaskType { get; set; }
    }

    public enum TaskType
    {
        None,
        Selection,
        Track,
    }
}

File: selections-loader/frontend/src/models/Task.ts
import {ModelBase} from '@/models/ModelBase';
import {TaskType} from '@/models/TaskType';

export class Task extends ModelBase {
    public name!: string

    public isFinished!: boolean

    public priority!: number

    public createDate!: Date

    public finishDate: Date | null = null

    public taskObjectId!: string

    public taskType!: TaskType

    public progress: number | null = null

    constructor(init?: Partial<Task>) {
        super();
        Object.assign(this, init);
    }
}

